resolvers/uploadYear.js

```diff
@@ -127,322 +55,66 @@ const uploadYear = async (_, { year, projects, data }) => {
-              rawOther.push(majors[k]);
-              found = true;
-            }
-          }
-        });
-      });
-    }
-
-    // Check to see if the hacker is a first time hacker
-    if (hackerData.data[i].hackathons === '0') {
-      firstTimeHackers++;
-    }
-
-    // Add the hacker email to the list of emails if the email field isn't blank
-    if (hackerData.data[i].email.length > 0) {
-      uniqueHackers.push(hackerData.data[i].email);
-    }
+  Object.values(hackerData.data).forEach((hacker) => {
+    helpers.populateFirstTimeHackers(metrics, hacker);
+    helpers.populateMajors(metrics, hacker);
+    helpers.populateIndependents(metrics.gender, hacker.gender);
+    helpers.populateIndependents(metrics.race, hacker.race);
+    helpers.populateIndependents(metrics.levelOfStudy, hacker.levelOfStudy);
+    helpers.populateIndependents(metrics.diet, hacker.diet);
+    helpers.populateIndependents(metrics.shirt, hacker.shirt);
   });
 
-  // Check to see if there are any previous added years
-  if (prevYearsData.data.allYears.length > 0) {
-    const prevYears = [];
-    // Get all the emails from the previous added years
-    Object.keys(prevYearsData.data.allYears).forEach((i) => {
-      prevYears.push({
-        year: prevYearsData.data.allYears[i].year,
-        metricId: prevYearsData.data.allYears[i].metrics.id,
-        uniqueHackers: prevYearsData.data.allYears[i].metrics.uniqueHackers.split(','),
-      });
-    });
-
-    // Go through email array
-    Object.keys(uniqueHackers).forEach((i) => {
-      // Go through previous added years checking for hacker emails
-      Object.keys(prevYears).forEach((j) => {
-        // Check to see if the email matches
-        if (prevYears[j].uniqueHackers.includes(uniqueHackers[i])) {
-          // Check the different cases a year may have
-          // If the year being added is less than the current previously added year
-          if (+year < +prevYears[j].year) {
-            // Splice the year from the previously added year
-            prevYears[j].uniqueHackers.splice(
-              prevYears[j].uniqueHackers.indexOf(uniqueHackers[i]),
-              1,
-            );
-          } else {
-            // Splice the year from the year being added
-            uniqueHackers.splice(i, 1);
-          }
-        }
-      });
-    });
-
-    // Update previous years metrics
-    await Object.keys(prevYears).forEach((i) => {
-      keystone.executeQuery(`
-        mutation {
-          updateMetric(
-            id: ${prevYears[i].metricsId},
-            data: {
-                uniqueHackers: ${prevYears[i].uniqueHackers.join(',')},
-            }
-          ){
-            uniqueHackers
-          }
-        }
-      `);
-    });
-  }
+  console.log("Here is your upload report! Make sure you look for any keys that aren't supposed to be there, this means that the parser found something that didn't match the templated keys.");
+  console.log(metrics);
```

Comment:
This will log a report of what we put into the database. Great for debugging and catching issues within CSV files.